packages/manager/core/application/src/ovh-routing.tsx

```diff
@@ -0,0 +1,136 @@
+import { lazy, useEffect, Fragment, Suspense } from 'react';
+import {
+  useLocation,
+  RouteObject,
+  createHashRouter,
+  Outlet,
+  LoaderFunction,
+  ActionFunction,
+  LoaderFunctionArgs,
+  ActionFunctionArgs,
+  useRouteError,
+} from 'react-router-dom';
+import {
+  generateRegularRoutes,
+  generatePreservedRoutes,
+} from 'generouted/core';
+
+import { useShell } from '.';
+
+type Element = () => JSX.Element;
+type Module = {
+  default: Element;
+  loader: LoaderFunction;
+  action: ActionFunction;
+  ErrorElement: Element;
+  breadcrumb: () => unknown;
+};
+
+function HidePreloader(): JSX.Element {
+  const shell = useShell();
+  useEffect(() => {
+    shell.ux.hidePreloader();
+  }, []);
+  return undefined;
+}
+
+function OvhContainerRoutingSync(): JSX.Element {
+  const location = useLocation();
+
+  const shell = useShell();
+  useEffect(() => {
+    shell.routing.stopListenForHashChange();
+  }, []);
+  useEffect(() => {
+    shell.routing.onHashChange();
+  }, [location]);
+  return undefined;
```

Comment:
**question:** Can you return `null` instead?

Reply:
same answer below